File: app/src/main/java/com/example/mg/masterdetail/data/IWeatherClient.java
package com.example.mg.masterdetail.data;

import com.example.mg.masterdetail.data.model.Weather10daysModel;
import com.example.mg.masterdetail.data.model.WeatherModel;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Path;

public interface IWeatherClient {
    String BASE_URL = "http://api.wunderground.com";
    String KEY = "24b6732692393b6b";
    String COUNTRY = "EG";
    String CITY = "Cairo";

    @Headers({
            "Accept: application/json",
            "Cache-Control: max-age=640000"
    })
    @GET("/api/{key}/conditions/q/{COUNTRY}/{CITY}.json")
    Single<WeatherModel> weatherForCity(@Path("key") String key,
                                        @Path("COUNTRY") String country,
                                        @Path("CITY") String city);

    @Headers({
            "Accept: application/json",
            "Cache-Control: max-age=640000"
    })
    @GET("/api/{key}/forecast10day/q/{COUNTRY}/{CITY}.json")
    Single<Weather10daysModel> weather10DaysForecast(@Path("key") String key,
                                                     @Path("COUNTRY") String country,
                                                     @Path("CITY") String city);
}

File: app/src/main/java/com/example/mg/masterdetail/data/IWeatherClient.kt
package com.example.mg.masterdetail.data

import com.example.mg.masterdetail.data.model.WeatherDaysModel
import com.example.mg.masterdetail.data.model.WeatherModel
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query

interface IWeatherClient {

    companion object {
        const val BASE_URL = "https://api.weatherbit.io/"
        const val KEY = "e802a22d8af64f098c9df4144aa467d1"
        const val CITY = "cairo"
    }

    @Headers("Accept: application/json", "Cache-Control: max-age=640000")
    @GET("v2.0/current")
    suspend fun weatherForCity(@Query("city") country: String,
                               @Query("key") key: String): Response<WeatherModel>

    @Headers("Accept: application/json", "Cache-Control: max-age=640000")
    @GET("v2.0/forecast/daily")
    suspend fun weatherDaysForCity(@Query("city") country: String,
                                   @Query("key") key: String): Response<WeatherDaysModel>

}